### DISCUSSION TITLE
Help! !!auth?.user is always null but if i use await auth i have a session

### DISCUSSION THREAD

**Discussion by Ganchinho** [nextauthjs/next-auth#9221] [Help]
Hello fellow programmers! This is not an issue, is more of a ask for help situation since i am a new dev.

I am trying to implement next auth v5 in my app, using next 14.

I have a route which is /new-product, and i when i visit this route i check if the user is logged in, and if it is not, it will return to the login page: 

--//--
auth.config.ts

import type { NextAuthConfig } from 'next-auth';
import { auth } from './auth';
 
export const authConfig = {
  pages: {
    signIn: '/login',
  },
  callbacks: {
    authorized({ auth, request: { nextUrl } }) {
      const isLoggedIn = auth?.user;
      const isOnNewProduct = nextUrl.pathname.startsWith('/new-product');
      //console.log(isOnNewProduct);
      console.log("Is Logged in: " + isLoggedIn);
      console.log("Is on New Product: " + isOnNewProduct);
      
      if (isOnNewProduct) {
        if (isLoggedIn) return true;
        return false; // Redirect unauthenticated users to login page
      } else if (isLoggedIn) {
          
        return Response.redirect(new URL('/new-product', nextUrl));
      } 
      return true;
    },
  }, 

  providers: [], // Add providers with an empty array for now
} satisfies NextAuthConfig;

--//--

In the login page i have a form with user and password, and i check after login my user session:

--//--
import LoginForm from "../components/login-form";
import { auth } from "../../auth";

export default async function Login (){
        
        const session = await auth();
        console.log("session: " + session);

        return  <LoginForm />
        
}
--//--

In the auth.config.ts file my auth?.user is allways null, even ater i log in, and in my login page, after i login my session object contains the user and when session expires... 

I do not understant why auth.user is allways null. Can someone help me please?

If there is a need for better clarification please let me know

Regards

**Comment by akash-mahmud** [comment]
I am facing the same problem

**Comment by yusufmafif** [comment]
Me too. Has this problem been solved?
